hyperspy/tests/io/test_zspy.py

```diff
@@ -0,0 +1,706 @@
+# -*- coding: utf-8 -*-
+# Copyright 2007-2021 The HyperSpy developers
+#
+# This file is part of  HyperSpy.
+#
+#  HyperSpy is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+#  HyperSpy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with  HyperSpy.  If not, see <http://www.gnu.org/licenses/>.
+
+import gc
+import os.path
+import sys
+import tempfile
+import time
+from os import remove
+
+import dask.array as da
+import numpy as np
+import pytest
+import zarr
+
+from hyperspy._signals.signal1d import Signal1D
+from hyperspy._signals.signal2d import Signal2D
+from hyperspy.datasets.example_signals import EDS_TEM_Spectrum
+from hyperspy.exceptions import VisibleDeprecationWarning
+from hyperspy.axes import AxesManager
+from hyperspy.io import load
+from hyperspy.misc.test_utils import assert_deep_almost_equal
+from hyperspy.misc.test_utils import sanitize_dict as san_dict
+from hyperspy.roi import Point2DROI
+from hyperspy.signal import BaseSignal
+from hyperspy.utils import markers
+import hyperspy.api as hs
+
+my_path = os.path.dirname(__file__)
+
+data = np.array([4066., 3996., 3932., 3923., 5602., 5288., 7234., 7809.,
+                 4710., 5015., 4366., 4524., 4832., 5474., 5718., 5034.,
+                 4651., 4613., 4637., 4429., 4217.])
+example1_original_metadata = {
+    'BEAMDIAM -nm': 100.0,
+    'BEAMKV   -kV': 120.0,
+    'CHOFFSET': -168.0,
+    'COLLANGLE-mR': 3.4,
+    'CONVANGLE-mR': 1.5,
+    'DATATYPE': 'XY',
+    'DATE': '01-OCT-1991',
+    'DWELLTIME-ms': 100.0,
+    'ELSDET': 'SERIAL',
+    'EMISSION -uA': 5.5,
+    'FORMAT': 'EMSA/MAS Spectral Data File',
+    'MAGCAM': 100.0,
+    'NCOLUMNS': 1.0,
+    'NPOINTS': 20.0,
+    'OFFSET': 520.13,
+    'OPERMODE': 'IMAG',
+    'OWNER': 'EMSA/MAS TASK FORCE',
+    'PROBECUR -nA': 12.345,
+    'SIGNALTYPE': 'ELS',
+    'THICKNESS-nm': 50.0,
+    'TIME': '12:00',
+    'TITLE': 'NIO EELS OK SHELL',
+    'VERSION': '1.0',
+    'XLABEL': 'Energy',
+    'XPERCHAN': 3.1,
+    'XUNITS': 'eV',
+    'YLABEL': 'Counts',
+    'YUNITS': 'Intensity'}
+
+
```

Comment:
It seems that many of these tests are repeated from `test_hdf5`, would it make sense to use `pytest.mark.parametrize` to check with `hspy` and `zspy`, where possible? It could easily get out of sync if we keep the duplicated tests.

Reply:
Good idea.  I'll do this.  It should also clean up some of the hspy tests as well